### ISSUE TITLE
URLSearchParams needs size

### ISSUE THREAD

**Issue by AndyOGo** [microsoft/TypeScript#54466]
# Bug Report

### 🔎 Search Terms

URLSearchParams

### 🕗 Version & Regression Information

**TypeScript Version:** `typescript@4.9.5`
- This is a crash
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about [`URLSearchParams.size`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/size)

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=5.2.0-dev.20230531#code/MYewdgziA2CmB00QHMAUZYHcAEBVASgDIDKsAhgE7AAWACpWQLYSoCU8EAlgF6ysBQ2bEA)

### 💻 Code

```ts
console.log(new URLSearchParams().size)
```

### 🙁 Actual behavior

> Property 'size' does not exist on type 'URLSearchParams'.ts(2339)

### 🙂 Expected behavior

No error

### Related bug:

https://github.com/microsoft/TypeScript/issues/38139

**Note:** Adding `{ "lib": ["dom.iterable"] }` to my `compilerOptions` does not help.

**Comment by AndyOGo** [comment]
Quick workaround is to extend `"dom.iterable"` locally, e.g.:

`URLSearchParams .d.ts`
````ts
interface URLSearchParams {
  size: number;
}
````

**Comment by fatcerberus** [comment]
Makes sense, according to MDN it seems like this only landed in Chrome this month.  TS tends to wait until at least 2 browser engines support an API before adding it to `dom`.

**Comment by bencmbrook** [comment]
This has rolled out quite widely since https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/size#browser_compatibility

**Comment by bencmbrook** [comment]
Interestingly it looks like it's showing up in `src/lib/dom.generated.d.ts` now

https://github.com/microsoft/TypeScript/blob/2170e6c6cc12f08bfa2975955da2f145e4be6101/src/lib/dom.generated.d.ts#L22555-L22557

https://github.com/microsoft/TypeScript/blob/main/src/lib/dom.generated.d.ts#L22555-L22557

**Comment by RyanCavanaugh** [comment]
`URLSearchParams.size` was added by [#54725](https://github.com/microsoft/TypeScript/pull/54725). The original expression reports TS2339 in 5.2.0-dev.20230621 and is accepted in 5.2.0-dev.20230622, TypeScript 6.0.3, and the current native compiler.
